packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts

```diff
@@ -223,48 +221,44 @@ export default util.createRule<Options, MessageIds>({
 
         context.report({
           fix: function* (fixer) {
+            // 1. isUnaryNegation - parent negation
+            // 2. literalBooleanInComparison - is compared to literal boolean
+            // 3. negated - is expression negated
+
+            const isUnaryNegation =
+              node.parent != null && nodeIsUnaryNegation(node.parent);
+
+            const shouldNegate = comparison.literalBooleanInComparison
+              ? !comparison.negated
+              : comparison.negated;
+
+            const mutatedNode = isUnaryNegation ? node.parent! : node;
+
             yield fixer.replaceText(
-              node,
+              mutatedNode,
               sourceCode.getText(comparison.expression),
             );
 
-            // if the expression `exp` isn't nullable, or we're comparing to `true`,
-            // we can just replace the entire comparison with `exp` or `!exp`
-            if (
-              !comparison.expressionIsNullableBoolean ||
-              comparison.literalBooleanInComparison
-            ) {
-              if (!comparison.forTruthy) {
-                yield fixer.insertTextBefore(node, '!');
-                if (!util.isStrongPrecedenceNode(comparison.expression)) {
-                  yield fixer.insertTextBefore(node, '(');
-                  yield fixer.insertTextAfter(node, ')');
-                }
-              }
-              return;
-            }
+            // if (XOR) `isUnaryNegation ^ literalBooleanInComparison ^ negated` is true - negate the expression
+            if (isUnaryNegation === shouldNegate) {
```

Comment:
i'm not sure if we want to but we could replace this with 
```ts
isUnaryNegation ^ comparison.literalBooleanInComparison ^ comparison.negated` 
```

and we could remove `shouldNegate` variable